MODULES-11188: trim physicalpath for iis_application resource

adrianiurca opened the pull request.

adrianiurca committed:
MODULES-11188: trim physicalpath for iis_application resource

chelnak commented:
@LukasAud is going to be pushing this PR through along with some new changes.

LukasAud committed:
[MODULES-11188] Additional utility changes

Prior to this commit, there was an attempt to fix an issue where puppet
could not handle a trailing '/' at the end of a physicalpath. While it
stopped users from creating resources with trailing '/', it did not fix
existing IIS entities that already had the trailing '/'.

This commit aims to complement the previous code by removing trailing
'/' in physicalpaths for existing resources.

BartoszBlizniak commented:
I have tested the latest change with the code supplied in the Jira ticket and everything works as expected now.

LukasAud committed:
[MODULES-11188] Fixing unit tests

Prior to this commit there were some unit tests that would fail due to
the additional checks inserted in the latest fix. Some of the previous
code would also not work as intended and lint was throwing warnings at
some formatting choices made.

This commit aims to provide a number of small fixes to deal with both,
lint warnings and small errors in previous code that would make unit
tests fail.

LukasAud committed:
[MODULES-11188] Updating README

Prior to this commit several fixes were implemented to deal with the
problem described in the JIRA ticket MODULES-11188. As a result of those
changes, the behaviour of the module regarding physicalpaths was
slighly altered.

This commit aims to update the README file to inform users about the
previously existing issue and the measures that the module will take to
deal with it.

david22swan commented on lib/puppet/type/iis_application.rb:
```diff
@@ -41,11 +41,21 @@
           fully qualified.'
-    validate do |value|
-      if value.nil? || value.empty?
+    # validate do |value|
```
should this commented out code still be here?

LukasAud replied:
Most comments left were from Adrian's original code. Probably not needed anymore

david22swan reviewed (approved):
LGTM

Failure on test's is a transient timeout
